### PR TITLE
Uptrain integration

### PR DESCRIPTION
**Pull request by Dominastorm**
# Description

This PR adds a tutorial and information about integration with UpTrain - LLMOps toolkit. 

## Type of Change

- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by Dominastorm** [comment]
Hey @hatianzhang, we were looking into creating a Callback Handler for UpTrain. We wished to get the query and retrieved context as well along with the response in the callback. Is there a way we can do this?

**Comment by hatianzhang** [comment]
I believe you could follow this https://github.com/run-llama/llama_index/blob/b10d26ef2abddbcc93f46bb314086fba5d0c5ff2/llama_index/callbacks/open_inference_callback.py to get those information for query, retrieved nodes, response

**Comment by hatianzhang** [comment]
@Dominastorm could you help resolve the conflicts?

File: docs/module_guides/evaluating/root.md [hunk 1]
```diff
@@ -49,6 +49,7 @@ The core retrieval evaluation steps revolve around the following:
 
 We also integrate with community evaluation tools.
 
+- [UpTrain](../../community/integrations/uptrain.md)
```
Comment (hatianzhang) [inline docs/module_guides/evaluating/root.md:52]: is it right url

Comment (Dominastorm) [reply docs/module_guides/evaluating/root.md:52]: Fixed it, thanks

**Comment by Dominastorm** [comment]
@hatianzhang I have resolved the conflicts. Working on the callback, will push it soon.

**Review by hatianzhang — APPROVED** [review]
LGTM

**Pull request merged**